engine/engine/src/engine.cpp

```diff
@@ -1016,6 +1022,15 @@ namespace dmEngine
         // Variables need to be declared up here due to the goto's
         bool has_host_mount = dmSys::GetEnv("DM_MOUNT_HOST") != 0;
 
+        engine->m_ResourceTypeContexts.Put(dmHashString64("goc"), engine->m_Register);
+        engine->m_ResourceTypeContexts.Put(dmHashString64("collectionc"), engine->m_Register);
+        engine->m_ResourceTypeContexts.Put(dmHashString64("scriptc"), engine->m_GOScriptContext);
+        engine->m_ResourceTypeContexts.Put(dmHashString64("luac"), &engine->m_ModuleContext);
+
+        fact_result = dmResource::RegisterTypes(engine->m_Factory, &engine->m_ResourceTypeContexts);
+        if (fact_result != dmResource::RESULT_OK)
+            goto bail;
+
         fact_result = dmGameObject::RegisterResourceTypes(engine->m_Factory, engine->m_Register, engine->m_GOScriptContext, &engine->m_ModuleContext);
```

Comment:
Leaving the gamesystem ones for a "cleanup" task later on, once we're satisfied this is good to work with.